src/main/java/com/anthonynsimon/url/URL.java

```diff
@@ -351,4 +385,62 @@ public URL resolveReference(URL ref) throws InvalidURLReferenceException {
                 .build();
     }
 
+
+    private static String mapToNullIfEmpty(String str) {
+        return str != null && !str.isEmpty() ? str : null;
+    }
+
+    /**
+     * Returns the full host (hostname:port), maps to null if none are set.
+     */
+    private static String mergeHostPortIfSet(String hostname, Integer port) {
+        StringBuilder sb = new StringBuilder();
+        boolean exists = false;
+        if (hostname != null) {
+            sb.append(hostname);
+            exists = true;
+        }
+        if (port != null) {
+            sb.append(":");
+            sb.append(port);
+            exists = true;
+        }
+        if (exists) {
+            return sb.toString();
+        }
+        return null;
+    }
+
+    /**
+     * Returns the hostname part of the host ('www.example.com' or '192.168.0.1' or '[fde2:d7de:302::]') if it exists.
+     */
+    private static String extractHostname(String host) {
+        if (host != null) {
+            int separator = host.lastIndexOf(":");
+            if (separator > -1) {
```

Comment:
Could you add `return host` in else statement here. This way hostname won't be null if url without port.

Reply:
Thanks for catching that! I fixed it on v0.4.2, let me know